app/commands/v2/put_content.rb

```diff
@@ -98,6 +99,19 @@ def create_links(edition)
         end
       end
 
+      def create_content_id_alias
+        content_id_alias = payload.dig(:details, :content_id_alias)
+        return unless content_id_alias
+
+        return if ContentIdAlias.find_by(name: content_id_alias, content_id: payload[:content_id])
+
+        if ContentIdAlias.find_by(name: content_id_alias)
+          raise CommandError.new(code: 422, message: "ContentIdAlias with name \"#{content_id_alias}\" exists for a different content ID.")
+        end
```

Comment:
Minor non-blocking suggestion: this will make two database queries that are almost identical. Could it be condensed into one?
```suggestion
        if content_id_alias = ContentIdAlias.find_by(name: content_id_alias)
          return if content_id_alias.content_id == payload[:content_id]
          
          raise CommandError.new(code: 422, message: "ContentIdAlias with name \"#{content_id_alias}\" exists for a different content ID.")
        end
```

Reply:
Yeah I did think about that - your suggestion looks good!